blog/2020-11-10-streaming-vitess-at-bolt.adoc

```diff
@@ -0,0 +1,181 @@
+
+Advantages:
+
+* It is compatible with VTGate.
+* It handles Vitess resharding operation better. During resharding operation, reads are simply automatically redirected (by VTGate) to the target shards. It won't generate any duplicates or lose any data.
+
+Disadvantages:
+
+* It is poll-based, meaning that the connector polls the database for new change events on a defined interval (typically every few seconds). This means that we would have a much higher latency, compared to the Debezium MySQL Connector.
+* Its offsets are managed by either the table's incremental primary key or one of the table's timestamp columns. If we use the timestamp column for offset, we'd have to create a secondary-index of the timestamp column for each table. This adds more constraints on our backend services. If we use the incremental primary key, we would miss the change events for row-updates because the primary key is simply not updated.
+* The topic name created by the JDBC connector doesn't include the table's schema name. Using the `topic.prefix` connector configuration would mean that we'll have one connector per schema. At Bolt, we have a large number of schemas, which means we would need to create a large number of JDBC Source Connectors.
+* At Bolt, our downstream applications are already set up to use Debezium's data formats and topic naming conventions, we'd need to change our downstream application's decoding logic to the new data formats.
+* Row deletes are not captured.
+
+=== Option 3: Using VStream gRPC
+
+VTGate exposes a gRPC service called VStream. It is a server-side streaming service. Any gRPC client can subscribe to the link:https://vitess.io/docs/concepts/vstream/[VStream] service to get a continuous stream of change events from the underlying MySQL instances. The change events that VStream emits have similar information to the MySQL binary logs of the underlying MySQL instances. A single VStream can even subscribe to multiple shards for a given keyspace, making it quite a convenient API to build CDC tools.
+
+Behind the scene, as shown in _Figure 2_, VStream reads change events from multiple link:https://vitess.io/docs/reference/programs/vttablet/[VTTablets], one VTTablet per shard. Therefore, it doesn’t send duplicates from multiple VTTablets for a given shard. Each VTTablet is a proxy to its MySQL instance. A typical topology would include one master VTTablet and its master MySQL instance, and multiple replica VTTablets, each of which is the proxy of its own replica MySQL instance. A VTTablet gets change events from its underlying MySQL instance and sends the change events back to VTGate, which in turn sends the change events back to VStream’s gRPC client.
```

Comment:
```suggestion
Behind the scene, as shown in _Figure 2_, VStream reads change events from multiple link:https://vitess.io/docs/reference/programs/vttablet/[VTTablets], one VTTablet per shard. Therefore, it doesn’t send duplicates from multiple VTTablets for a given shard. Each VTTablet is a proxy to its MySQL instance. A typical topology would include one primary VTTablet and its corresponding MySQL instance, and multiple replica VTTablets, each of which is the proxy of its own replica MySQL instance. A VTTablet gets change events from its underlying MySQL instance and sends the change events back to VTGate, which in turn sends the change events back to VStream’s gRPC client.
```

Reply:
Shall we keep the `master VTTablet` part since it is a Vitess terminology: https://vitess.io/docs/concepts/tablet/#tablet-types